lms/djangoapps/learner_dashboard/tests/test_views.py

```diff
@@ -93,10 +94,15 @@ def test_program_discussion_disabled(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data, expected_data)
 
-    def test_api_returns_discussions_iframe(self):
+    @ddt.data(True, False)
+    def test_api_returns_discussions_iframe(self, staff):
         """
-        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data
+        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data for staff
+        and non staff. In case of non staff user must be be enrolled in program.
```

Comment:
Tests that API returns LTI discussion iframe when the provider is properly configured. The scenario is true for both staff and nonstaff users.

Reply:
The scenario is not true for both staff and nonstaff. non staff requires an additional enrolment condition.

Reply:
I see got it. Then I think its simpler to break the test into two? what do you think?

Reply:
The current test is simpler. In case of staff user I create a new user that no enrolment and test with that. The default user is a non staff user with an enrolment